opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1207,7 +1207,7 @@ namespace Opm
         const WellControls* wc = well_controls_;
         const double* distr = well_controls_get_current_distr(wc);
 
-        if (well_controls_get_current_type(wc) == RESERVOIR_RATE) {
+        if (well_controls_get_current(wc) != -1 && well_controls_get_current_type(wc) == RESERVOIR_RATE) {
```

Comment:
Maybe we can add a function to indicate the current control is valid or not for the control. It is basically equivalent to whether the `current  >=0`.   Anyway, it is just nitpicking and might just be easier to read.